zapcore/json_encoder_test.go

```diff
@@ -171,3 +172,65 @@ func TestJSONEmptyConfig(t *testing.T) {
 		})
 	}
 }
+
+// Encodes any object into empty json '{}'
+type emptyReflectedEncoder struct {
+	writer io.Writer
+}
+
+func (enc *emptyReflectedEncoder) Encode(obj interface{}) error {
+	_, err := enc.writer.Write([]byte("{}"))
+	return err
+}
+
+func TestJSONCustomReflectionEncoder(t *testing.T) {
+	tests := []struct {
+		name     string
+		field    zapcore.Field
+		expected string
+	}{
+		{
+			name: "encode custom map object",
+			field: zapcore.Field{
+				Key:  "data",
+				Type: zapcore.ReflectType,
+				Interface: map[string]interface{}{
+					"foo": "hello",
+					"bar": 1111,
+				},
+			},
+			expected: `{"data":{}}`,
+		},
+		{
+			name: "encode nil object",
+			field: zapcore.Field{
+				Key:  "data",
+				Type: zapcore.ReflectType,
+			},
+			expected: `{"data":null}`,
+		},
+	}
+
+	enc := zapcore.NewJSONEncoder(zapcore.EncoderConfig{
+		NewReflectedEncoder: func(writer io.Writer) zapcore.ReflectedEncoder {
+			return &emptyReflectedEncoder{
+				writer: writer,
+			}
+		},
+	})
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			buf, err := enc.EncodeEntry(zapcore.Entry{
```

Comment:
```suggestion
	for _, tt := range tests {
	        tt := tt
		t.Run(tt.name, func(t *testing.T) {
		        t.Parallel()
			buf, err := enc.EncodeEntry(zapcore.Entry{
```

Reply:
Oh, we can do that, although for that the encoder has to be instantiated per-test.
(Encoder isn't safe for concurrent use.)
I'll do it.